### PR TITLE
Add /api/update-person functionality

### LINKED ISSUE #50: Frontend must be able to update the person
The /api/update-person servlet should take in the updated person and an update mask, then perform the required operations and save the updated person in the database.

### PR DESCRIPTION
**Pull request by JosephBushagour**
The frontend can now update a person in the database by sending a request to /api/update-person with the representation of the person to be updated in the database and an optional update mask.

### CODE REVIEW

File: server/src/main/java/com/google/coffeehouse/servlets/UpdatePersonServlet.java [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+   */
+  public static final String BODY_ERROR = "- unable to parse body.";
+  /** Message to be logged when the body of the POST request cannot be parsed. */
+  public static final String LOG_BODY_ERROR_MESSAGE =
+      "Body unable to be parsed in UpdatePersonServlet: ";
+  /** 
+   * The error string sent by the response object in doPost when the body of the 
+   * POST request does not have a "person" field.
+   */
+  public static final String NO_PERSON_ERROR = "No \"person\" found in JSON.";
```
Comment (KeatonTech) [inline server/src/main/java/com/google/coffeehouse/servlets/UpdatePersonServlet.java]: nit: Consider using Java `String.format` here, so the string would be "No \"%s\" found in JSON." and then you could use String.format to pass in the field name that's missing

File: server/src/main/java/com/google/coffeehouse/servlets/UpdatePersonServlet.java [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+    this.storageHandler = storageHandler;
+  }
+
+  /** 
+   * Explicit default constructor used for instantiating the servlet when not testing.
+   */
+  public UpdatePersonServlet() {
+    super();
+    this.storageHandler = new StorageHandlerApi();
+  }
```
Comment (KeatonTech) [inline server/src/main/java/com/google/coffeehouse/servlets/UpdatePersonServlet.java:74]: I'm not sure we want this constructor. Shouldn't there only be one instance of StorageHandlerApi in the app?

File: server/src/main/java/com/google/coffeehouse/servlets/UpdatePersonServlet.java [hunk 3]
```diff
@@ -0,0 +1,142 @@
...
+   *     parameter) this object will send a "400 Bad Request error"
+   * @throws IOException if an input or output error is detected when the servlet handles the request
+   */
+  @Override
+  public void doPost(HttpServletRequest request, HttpServletResponse response) throws IOException {
+    Person personToUpdate;
+    try {
+      // Get the update mask / updated Person from the post request.
+      Map mappedRequest = gson.fromJson(request.getReader(), Map.class);
+      Map personInfo = (Map) mappedRequest.getOrDefault(PERSON_FIELD_NAME, null);
```
Comment (KeatonTech) [inline server/src/main/java/com/google/coffeehouse/servlets/UpdatePersonServlet.java:97]: nit: Technically this is equivalent to just calling `.get`, which defaults to null

File: server/src/main/java/com/google/coffeehouse/servlets/UpdatePersonServlet.java [hunk 4]
```diff
@@ -0,0 +1,142 @@
...
+      JsonObject personToUpdateJson = gson.toJsonTree(personToUpdate).getAsJsonObject();
+
+      // Merge the updates from the request with the Person from the database.
+      List<String> mask = rawUpdateMask == null 
+          ? updateableFields
+          : Arrays.asList(rawUpdateMask.split(","));
+
+      mask.stream()
+          .filter(updateableFields::contains)
+          .forEach(elem -> personToUpdateJson.addProperty(elem, (String) personInfo.get(elem)));
```
Comment (KeatonTech) [inline server/src/main/java/com/google/coffeehouse/servlets/UpdatePersonServlet.java:119]: Nice :)

**Review by KeatonTech — CHANGES_REQUESTED** [review]

File: server/src/main/java/com/google/coffeehouse/servlets/UpdatePersonServlet.java (hunk 3 above)
Comment (JosephBushagour) [reply server/src/main/java/com/google/coffeehouse/servlets/UpdatePersonServlet.java:97]: Haha I realized this too late. All of the codebase uses getOrDefault() so I figured I'd stay consistent even if it's a little redundant.

File: server/src/main/java/com/google/coffeehouse/servlets/UpdatePersonServlet.java (hunk 2 above)
Comment (JosephBushagour) [reply server/src/main/java/com/google/coffeehouse/servlets/UpdatePersonServlet.java:74]: All StorageHandlerApi does is call static methods on StorageHandlerApi with a private static final dbClient that it has already created. The reason it was passed in to the constructor was solely to mock it easier (previously, we just referenced it statically like `StorageHandlerApi.fetchPersonFromId()`). Being honest I'm not sure how much of a problem this is, if it is a big problem, then we can reference it statically instead, I just don't know how to mock that out in unit tests.

File: server/src/main/java/com/google/coffeehouse/servlets/UpdatePersonServlet.java (hunk 1 above)
Comment (JosephBushagour) [reply server/src/main/java/com/google/coffeehouse/servlets/UpdatePersonServlet.java]: Good idea! Has been added and pushed.

File: server/src/main/java/com/google/coffeehouse/servlets/UpdatePersonServlet.java (hunk 2 above)
Comment (KeatonTech) [reply server/src/main/java/com/google/coffeehouse/servlets/UpdatePersonServlet.java:74]: Hmm, ok. Seems like kind of an odd pattern but I can't really think of anything better just given the way Servlets work so, SG

File: server/src/main/java/com/google/coffeehouse/servlets/UpdatePersonServlet.java (hunk 3 above)
Comment (KeatonTech) [reply server/src/main/java/com/google/coffeehouse/servlets/UpdatePersonServlet.java:97]: Ok SG

**Review by KeatonTech — APPROVED** [review]

**Review by linamontes10-zz — APPROVED** [review]
Looks good!

**Review by jimmycolema — APPROVED** [review]
lgtm!

**Review by SarahGB — APPROVED** [review]

**Pull request merged**